.github/workflows/build_and_publish.yml

```
@@ -4,6 +4,8 @@ on:
   push:
     branches:
       [ main ]
+  pull_request:
+    types: [ opened, synchronize, reopened, ready_for_review ]
```

You can drop `ready_for_review`. We do it elsewhere because we are preventing builds for draft PRs. But in your case, it's not worth going through the added complexity.
```suggestion
    types: [ opened, synchronize, reopened ]
```